jobs/jobs_watcher.go

```diff
@@ -72,138 +72,11 @@ func (watcher *Watcher) PollAndNotify() {
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeLdapSync {
-			if watcher.workers.LdapSync != nil {
-				select {
-				case watcher.workers.LdapSync.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeMigrations {
-			if watcher.workers.Migrations != nil {
-				select {
-				case watcher.workers.Migrations.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypePlugins {
-			if watcher.workers.Plugins != nil {
-				select {
-				case watcher.workers.Plugins.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeExpiryNotify {
-			if watcher.workers.ExpiryNotify != nil {
-				select {
-				case watcher.workers.ExpiryNotify.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeProductNotices {
-			if watcher.workers.ProductNotices != nil {
-				select {
-				case watcher.workers.ProductNotices.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeActiveUsers {
-			if watcher.workers.ActiveUsers != nil {
-				select {
-				case watcher.workers.ActiveUsers.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeImportProcess {
-			if watcher.workers.ImportProcess != nil {
-				select {
-				case watcher.workers.ImportProcess.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeImportDelete {
-			if watcher.workers.ImportDelete != nil {
-				select {
-				case watcher.workers.ImportDelete.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeExportProcess {
-			if watcher.workers.ExportProcess != nil {
-				select {
-				case watcher.workers.ExportProcess.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeExportDelete {
-			if watcher.workers.ExportDelete != nil {
-				select {
-				case watcher.workers.ExportDelete.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeCloud {
-			if watcher.workers.Cloud != nil {
-				select {
-				case watcher.workers.Cloud.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeResendInvitationEmail {
-			if watcher.workers.ResendInvitationEmail != nil {
-				select {
-				case watcher.workers.ResendInvitationEmail.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeExtractContent {
-			if watcher.workers.ExtractContent != nil {
-				select {
-				case watcher.workers.ExtractContent.JobChannel() <- *job:
-				default:
-				}
-			}
-		} else if job.Type == model.JobTypeFixChannelUnreadsForCRT {
-			if watcher.workers.FixCRTChannelUnreads != nil {
-				select {
-				case watcher.workers.FixCRTChannelUnreads.JobChannel() <- *job:
-				default:
-				}
+		worker := watcher.workers.Get(job.Type)
+		if worker != nil {
+			select {
+			case worker.JobChannel() <- *job:
+			default:
```

Comment:
Do we need to be concerned about jobs of the same type being executed in the order they are returned from the DB? If I'm reading the code correctly, when a worker is fetched from `watcher.workers.Get` and cannot accept a job immediately in its queue for whatever reason, the job is ignored and we move on the to next job, which may be of the same type and come after the ignored job in the result set.

Reply:
I don't have an strong opinion here, I just make it equivalent to the previous existing implementation. I think more about it in a second iteration. We can keep discussing it here, but I don't think we need to fix this as part of this PR.